lib/AST/Decl.cpp

```diff
@@ -2688,6 +2688,7 @@ mapSignatureExtInfo(AnyFunctionType::ExtInfo info,
     return AnyFunctionType::ExtInfo();
   return AnyFunctionType::ExtInfo()
       .withRepresentation(info.getRepresentation())
+      .withClangFunctionType(info.getClangFunctionType())
```

Comment:
This is a bit dissatisfying, I'm considering having a method on ExtInfo to handle this, so we can avoiding having a `getClangFunctionType()` method.